From 7988aa515c2a609c910c65076aaca34c64d8d63a Mon Sep 17 00:00:00 2001
From: czietz <czietz@gmx.net>
Date: Fri, 4 Jan 2019 17:03:47 +0000
Subject: [PATCH] Zero-terminate partition ID returned by XHInqDev2.

The XHDI 1.30 specification (German original version) mandates this.
This detail was omitted from the English translation.
---
 bios/xhdi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bios/xhdi.c b/bios/xhdi.c
index f7089dd2..2f7f6ca5 100644
--- a/bios/xhdi.c
+++ b/bios/xhdi.c
@@ -126,6 +126,7 @@ static long XHInqDev2(UWORD drv, UWORD *major, UWORD *minor, ULONG *start,
         partid[0] = blkdev[drv].id[0];
         partid[1] = blkdev[drv].id[1];
         partid[2] = blkdev[drv].id[2];
+        partid[3] = 0;
     }
 
    return E_OK;
-- 
2.17.0.windows.1

